pynitrokey/helpers.py

```diff
@@ -34,6 +36,43 @@
 STDOUT_PRINT = True
 
 
+def normalize_parameters(s: str) -> list[str]:
+    """Helper function to normalize different writing of parameters
+
+    `s`: `str` parameter to normalize
+
+    Returns:
+        `list of str`: normalized form of `parameter`
+    """
+    if s.startswith("--"):
+        return s.split("=", maxsplit=1)
+    else:
+        return [s]
+
+
+def filter_sensitive_parameters(parameters: list[str]) -> list[str]:
+    """Helper function to remove sensitive parameters
+
+    `parameters`: `list of str`
+
+    Returns:
+        `list of str`: `parameters` without sensitive values listed
+                       in `pynitrokey.confconsts.CLI_LOG_BLACKLIST`
+    """
+    parameters = list(
+        chain.from_iterable(
+            [normalize_parameters(parameter) for parameter in parameters]
+        )
+    )
+
+    for sensitive_parameter, parameter_length in CLI_LOG_BLACKLIST:
+        if sensitive_parameter in parameters:
+            sensitive_index = parameters.index(sensitive_parameter)
+            for i in range(sensitive_index + 1, sensitive_index + 1 + parameter_length):
+                parameters[i] = "[redacted]"
```

Comment:
This can fail if an argument is accidentally set twice (`nitropy nk3 test --pin 123456 --pin 123456`: second pin is not redacted) or if an argument is used incorrectly (`nitropy nk3 test --pin`: causes an error).  What do you think about something like this?

```
redact_count = 0
for i in range(len(parameters)):
    if redact_count > 0:
        parameters[i] = "[redacted]"
        redact_count -= 1
    elif parameters[i] in CLI_LOG_BLACKLIST:
        redact_count = CLI_LOG_BLACKLIST
```

This would mean that `CLI_LOG_BLACKLIST` should be a `dict[str, int]` instead of a `list[tuple[str, int]]`.

Or if you just use a `list[str]`:

```
redact_next = False
for i in range(len(parameters)):
    if redact_next:
        parameters[i] = "[redacted]"
        redact_next = False
    else:
        redact_next = parameters[i] in CLI_LOG_BLACKLIST
```